h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -151,8 +163,47 @@ public void setGroupBy(ArrayList<Expression> group) {
         return group;
     }
 
-    public HashMap<Expression, Object> getCurrentGroup() {
-        return currentGroup;
+    /**
+     * Is there currently a group-by active
+     */
+    public boolean isCurrentGroup() {
+        return currentGroupByExprData != null;
+    }
+
+    /**
+     * Get the group-by data for the current group and the passed in expression.
+     */
+    public Object getCurrentGroupExprData(Expression expr) {
+        Integer index = exprToIndexInGroupByData.get(expr);
+        if (index == null) {
+            return null;
+        }
+        return currentGroupByExprData[index];
+    }
+
+    /**
+     * Set the group-by data for the current group and the passed in expression.
+     */
+    public void setCurrentGroupExprData(Expression expr, Object obj) {
+        Integer index = exprToIndexInGroupByData.get(expr);
+        if (index != null) {
+            if (currentGroupByExprData[index] != null) {
+                throw DbException.throwInternalError();
+            }
```

Comment:
Looks like `assert` should be used instead. All existing callers of this method ensure that current value is `null` by theirself.